### PR TITLE
add docker support to katello_repository

### PR DESCRIPTION
**Pull request by sean797**
Also required for the newly incoming content_view_filter module tests.

 Apologies for the onslaught of PRs :rocket:

### CODE REVIEW

File: modules/katello_repository.py [hunk 1]
```diff
@@ -142,8 +153,9 @@ def main():
             product=dict(required=True),
             organization=dict(required=True),
             name=dict(required=True),
-            content_type=dict(required=True),
+            content_type=dict(required=True, choices=['docker', 'ostree', 'yum', 'puppet', 'file', 'deb']),
             url=dict(),
+            docker_upstream_name=dict(),
```
Comment (mdellweg) [inline modules/katello_repository.py]: Should there be a test, that this is only used in combination with `content_type='docker'`?

Comment (sean797) [reply modules/katello_repository.py]: I guess so, but only a pleb would do that :laughing:

File: modules/katello_repository.py [hunk 2]
```diff
@@ -162,6 +174,9 @@ def main():
     password = entity_dict.pop('password')
     state = entity_dict.pop('state')
 
+    if entity_dict['content_type'] != 'docker' and 'docker_upstream_name' in entity_dict:
+        module.fail_json(msg="docker_upstream_name should not be set unless content_type: docker")
+
```
Comment (mdellweg) [inline modules/katello_repository.py]: :+1:

File: modules/katello_repository.py [hunk 3]
```diff
@@ -61,10 +61,20 @@
         description:
             - The content type of the repository (e.g. yum)
         required: true
+        choices:
+            - docker
+            - ostree
+            - yum
+            - puppet
+            - file
+            - deb
```
Comment (ehelms) [inline modules/katello_repository.py]: I'd opt for ordering these based on typical usage or alphabetically

**Comment by sean797** [comment]
updated

**Pull request merged**